airo-robots/airo_robots/manipulators/position_manipulator.py

```diff
@@ -231,3 +295,58 @@ def _assert_joint_configuration_is_valid(self, joint_configuration: JointConfigu
             raise ValueError(
                 f"joint configuration {joint_configuration} is not reachable, could be because of kinematic constraints or safety constraints"
             )
+
+    def _assert_joint_configuration_nearby(
+        self, joint_configuration: JointConfigurationType, absolute_angle_tolerance=np.radians(1.0)
+    ) -> None:
+        """Assert that a joint configuration is nearby the current configuration.
+
+        Args:
+            joint_configuration: the configuration that should be nearby the current configuration.
+            absolute_angle_tolerance: the absolute tolerance for the comparison.
+
+        Raises:
+            ValueError: If the joint configuration is not nearby the current configuration."""
+        current_configuration = self.get_joint_configuration()
+        if (
+            not np.isclose(joint_configuration, current_configuration, atol=absolute_angle_tolerance, rtol=0.0)
+            .all()
+            .item()
+        ):
+            raise ValueError(
+                f"joint configuration {joint_configuration} is not nearby the current configuration {current_configuration}"
+            )
+
+    def _assert_joint_trajectory_start_time_is_zero(self, joint_trajectory: SingleArmTrajectory) -> None:
+        if joint_trajectory.times[0] != 0.0:
+            raise ValueError("joint trajectory should start at time 0.0")
+
+    def _assert_joint_trajectory_is_executable(self, joint_trajectory: SingleArmTrajectory) -> None:
+        self._assert_joint_trajectory_start_time_is_zero(joint_trajectory)
+
+        if joint_trajectory.path.positions is None:
+            raise ValueError("joint trajectory should contain joint positions")
+
+        self._assert_joint_configuration_nearby(joint_trajectory.path.positions[0])
+
+        if joint_trajectory.path.velocities is not None:
```

Comment:
might want to calculate the leading axis velocities and check if they are not too high? 

Is the velocity profile used currently? Bc the servoJ does not implement it iirc?